### PR TITLE
Publish the list of partition ids on TABLE_PARTITION_UPDATE event

### PR DESCRIPTION
**Pull request by rsyncv**

### CODE REVIEW

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java [hunk 1]
```diff
@@ -174,6 +179,25 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
...
+    /**
+     * get partition name list from list of partitionDtos. The returned list is capped at
+     * the first PARTS_UPDATED_LIST_MAX_SIZE elements, if there are more than that number of elements
+     * in the input then the return is empty which serves as a signal that complete list cannot be included
+     *
+     * @param partitionDtos partition DTOs
+     * @return list of partition ids from the input list
+     */
+    protected static List<String> getPartitionNameListFromDtos(final List<PartitionDto> partitionDtos) {
+        if (partitionDtos.size() > PARTS_UPDATED_LIST_MAX_SIZE) {
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java:191]: We should make PARTS_UPDATED_LIST_MAX_SIZE configurable.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java [hunk 2]
```diff
@@ -174,6 +179,25 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
...
+            // empty list signals
+            return Collections.emptyList();
+        }
+        return partitionDtos.stream()
+            .map(PartitionDto::getName)
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java:196]: Minor: We can as well make one mapper call with `dto -> dto.getName().getPartitionName()`

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 1 above)
Comment (rsyncv) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java:191]: Makes sense

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 2 above)
Comment (rsyncv) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java:196]: Ok

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java [hunk 3]
```diff
@@ -174,6 +179,25 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
             .collect(Collectors.toList());
     }
 
+    /**
+     * get partition name list from list of partitionDtos. The returned list is capped at
+     * the first PARTS_UPDATED_LIST_MAX_SIZE elements, if there are more than that number of elements
```
Comment (raveeram) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java:184]: should we just call this PARTITIONS_UPDATED_LIST_MAX_SIZE

**Review by raveeram — DISMISSED** [review]

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 3 above)
Comment (rsyncv) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java:184]: ok changed

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 1 above)
Comment (rsyncv) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: Need to get handle on how to properly do this, will break into separate change

**Review by ArtfulCoder — APPROVED** [review]
Thank you!
The max partition size will be made configurable in a future PR.

**Pull request merged**